node/pkg/websocketfetcher/providers/gopax/gopax.go

```diff
@@ -0,0 +1,118 @@
+package gopax
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"bisonai.com/orakl/node/pkg/websocketfetcher/common"
+	"bisonai.com/orakl/node/pkg/wss"
+	"github.com/rs/zerolog/log"
+	"nhooyr.io/websocket"
+)
+
+type GopaxFetcher common.Fetcher
+
+func New(ctx context.Context, opts ...common.FetcherOption) (common.FetcherInterface, error) {
+	config := &common.FetcherConfig{}
+	for _, opt := range opts {
+		opt(config)
+	}
+
+	fetcher := &GopaxFetcher{}
+	fetcher.FeedMap = config.FeedMaps.Separated
+	fetcher.FeedDataBuffer = config.FeedDataBuffer
+
+	subscription := []any{
+		Subscription{
+			Name: "SubscribeToTickers",
+		},
+	}
+
+	ws, err := wss.NewWebsocketHelper(ctx,
+		wss.WithEndpoint(URL),
+		wss.WithSubscriptions(subscription),
+		wss.WithProxyUrl(config.Proxy),
+		wss.WithReadLimit(327690),
+		wss.WithCustomReadFunc(fetcher.customReadFunc),
+	)
+	if err != nil {
+		log.Error().Str("Player", "Gopax").Err(err).Msg("error in gopax.New")
+		return nil, err
+	}
+	fetcher.Ws = ws
+	return fetcher, nil
+}
+
+func (f *GopaxFetcher) handleMessage(ctx context.Context, message map[string]any) error {
+	response, err := common.MessageToStruct[Response](message)
+	if err != nil {
+		return err
+	}
+
+	if response.Name == "SubscribeToTickers" {
+		var initialResponse InitialResponse
+		err := json.Unmarshal(response.Object, &initialResponse)
+		if err != nil {
+			return fmt.Errorf("failed to parse initial response: %v", response)
+		}
+		feedDataList := InitialResponseToFeedData(initialResponse, f.FeedMap)
+
+		for _, feedData := range feedDataList {
+			f.FeedDataBuffer <- *feedData
+		}
+		return nil
+	}
+
+	if response.Name == "TickerEvent" {
+		var tickers Tickers
+		err := json.Unmarshal(response.Object, &tickers)
+		if err != nil {
+			return fmt.Errorf("failed to parse response: %v", response)
+		}
+		for _, ticker := range tickers {
+			feedData, err := TickerToFeedData(ticker, f.FeedMap)
+			if err != nil {
+				if err.Error() == "feed not found" {
```

Comment:
how about moving this error to a list of other errors, i think it was in `sentinel`?

Reply:
yh I was feeling bothered to do that 🥹
but yes, will do